greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/SearchTermBuilder.java

```diff
@@ -110,13 +130,59 @@ public static SearchTermBuilder create(final SearchKey key) {
             case OR:
                 builder = createORTermBuilder();
                 break;
+            case SINCE:
+                builder = createReceivedDateTermBuilder(ComparisonTerm.GE);
+                break;
+            case ON:
+                builder = createReceivedDateTermBuilder(ComparisonTerm.EQ);
+                break;
+            case BEFORE:
+                builder = createReceivedDateTermBuilder(ComparisonTerm.LT);
+                break;
+            case SENTSINCE:
+                builder = createSentDateTermBuilder(ComparisonTerm.GE);
+                break;
+            case SENTON:
+                builder = createSentDateTermBuilder(ComparisonTerm.EQ);
+                break;
+            case SENTBEFORE:
+                builder = createSentDateTermBuilder(ComparisonTerm.LT);
+                break;             
             default:
                 throw new IllegalStateException("Unsupported search term '" + key + '\'');
         }
         builder.setSearchKey(key);
         return builder;
     }
-    
+
+    private static SearchTermBuilder createSentDateTermBuilder(final int searchTerm) {
+        return new SearchTermBuilder() {
+            @Override
+            public SearchTerm build() {
+                return new SentDateTerm(searchTerm, parseDate(getParameters()));
+            }
+        };
+    }
+
+    private static SearchTermBuilder createReceivedDateTermBuilder(final int searchTerm) {
+        return new SearchTermBuilder() {
+            @Override
+            public SearchTerm build() {
+                return new ReceivedDateTerm(searchTerm, parseDate(getParameters()));
+            }
+        };
+    }
+
+    private static Date parseDate(List<String> parameters) {
+        DateFormat df = new SimpleDateFormat("dd-MMM-yyyy");
+        try {
+            Date d = df.parse(parameters.get(0));
+            LOGGER.debug("Using date '{}'.", d);
+            return d;
+        } catch (ParseException e) {
+            throw new RuntimeException(e);
```

Comment:
More friendly exception with not parseable value, eg.:
new IllegalArgumentException("Can not parse date "+parameters.get(0), e)